build.rs

```diff
@@ -90,10 +93,15 @@ fn main() {
 
     let musl_v1_2_3 = env::var("RUST_LIBC_UNSTABLE_MUSL_V1_2_3").is_ok();
     println!("cargo:rerun-if-env-changed=RUST_LIBC_UNSTABLE_MUSL_V1_2_3");
+    let musl32_time64 = env::var("RUST_LIBC_UNSTABLE_MUSL_TIME64").is_ok();
+    println!("cargo:rerun-if-env-changed=RUST_LIBC_UNSTABLE_MUSL_TIME64");
     // loongarch64 and ohos have already updated
-    if musl_v1_2_3 || target_arch == "loongarch64" || target_env == "ohos" {
-        // FIXME(musl): enable time64 api as well
+    if ((musl_v1_2_3 || target_arch == "loongarch64") && target_env == "musl") || target_env == "ohos" {
         set_cfg("musl_v1_2_3");
+        if musl32_time64 && MUSL_TIME64_ARCHS.contains(&target_arch.as_str()) {
+            set_cfg("musl32_time64");
+            set_cfg("linux_time_bits64");
+        }
```

Comment:
(first commit) don't check for architectures explicitly, use `target_pointer_width` but opt out on specific arches if needed (similar to the GNU config)